clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -3854,6 +3854,48 @@ def MemCpyOp : CIR_Op<"libc.memcpy"> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// MemMoveOp
+//===----------------------------------------------------------------------===//
+
+def MemMoveOp : CIR_Op<"libc.memmove"> {
```

Comment:
You could use `AllTypesMatch` to constrain that the types of `$dst` and `$src` are the same:

```suggestion
def MemMoveOp : CIR_Op<"libc.memmove", [AllTypesMatch<["dst", "src"]>]> {
```